components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/AuthenticationRequestBuilder.java

```diff
@@ -100,6 +107,7 @@ private Event getEvent(AuthenticationContext context) throws ActionExecutionRequ
                 context.getServiceProviderName()));
         eventBuilder.currentStepIndex(context.getCurrentStep());
         eventBuilder.authenticatedSteps(getAuthenticatedStepsForEventBuilder(context));
+        eventBuilder.request(getRequest(request));
```

Comment:
How does headers or params gets filtered from that's configured ?

Reply:
Filtering logic is in the action executor core.
https://github.com/wso2/carbon-identity-framework/blob/5a6b0ef73698bce76636d2fa11a645fc30715985/components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/internal/service/impl/ActionExecutorServiceImpl.java#L220C13-L233C43

Didn't add the filtering part here due to avoid executing redundant logics.